cuQuantum SDK 1.0 release has different function signatures

#### Issue description

The 1.0 release of the cuQuantum SDK has different function signatures than the pre-release version, making the current release of Lightning GPU not work against the 1.0 cuQuantum release. Given the prerelease version is no longer available for download, users will be unable to have a working `lightning.gpu` device.

* *Expected behavior:* Lightning GPU wheels should be compilable against the cuQuantum 1.0 SDK release. Lightning CPU wheels should work against the custatevec.so library without any missing symbols.

* *Actual behavior:* `pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp:623:52: error: 'custatevecApplyExp' was not declared in this scope; did you mean 'custatevecApplyMatrix'?`

* *Reproduces how often:* Always with 1.0 release of cuQuantum SDK.

--- a/.github/CHANGELOG.md
+++ b/.github/CHANGELOG.md
@@ -1,3 +1,25 @@
+# Release 0.22.0
+
+* Formal release with NVIDIA cuQuantum SDK 1.0 support. 
+
+### Improvements
+
+* Release semantic versioning matches PennyLane current release versioning.
+[(#6)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/6)
+
+### Bug fixes
+
+* This release updates the cuQuantum function calls to match the SDK 1.0 release.
+[(#6)](https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/6)
+
+
+### Contributors
+
+This release contains contributions from (in alphabetical order):
+Ali Asadi, and Lee James O'Riordan
+
+
+
 # Release 0.1.0
 
  * Initial release. The PennyLane-Lightning-GPU device adds support for CUDA-capable GPU simulation through use of the NVIDIA cuQuantum SDK.

--- a/.github/workflows/build_wheel_manylinux2014.yml
+++ b/.github/workflows/build_wheel_manylinux2014.yml
@@ -17,11 +17,11 @@ env:
     yum-config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/cuda-rhel7.repo -y
     yum clean all
     yum -y install cuda cmake git openssh wget
-    wget -P /project https://developer.download.nvidia.com/compute/cuquantum/redist/cuquantum/linux-x86_64/cuquantum-linux-x86_64-0.1.0.30-archive.tar.xz
-    tar xvf /project/cuquantum-linux-x86_64-0.1.0.30-archive.tar.xz -C /project
+    wget -P /project https://developer.download.nvidia.com/compute/cuquantum/redist/cuquantum/linux-x86_64/cuquantum-linux-x86_64-22.03.0.40-archive.tar.xz
+    tar xvf /project/cuquantum-linux-x86_64-22.03.0.40-archive.tar.xz -C /project
 
   # ensure nvcc is available
-  CIBW_ENVIRONMENT: PATH=$PATH:/usr/local/cuda/bin LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/project/cuquantum-linux-x86_64-0.1.0.30-archive/lib CUQUANTUM_SDK=/project/cuquantum-linux-x86_64-0.1.0.30-archive
+  CIBW_ENVIRONMENT: PATH=$PATH:/usr/local/cuda/bin LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/project/cuquantum-linux-x86_64-22.03.0.40-archive/lib CUQUANTUM_SDK=/project/cuquantum-linux-x86_64-22.03.0.40-archive
 
   CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014
   CIBW_REPAIR_WHEEL_COMMAND_LINUX: "./docker/auditwheel repair -w {dest_dir} {wheel}"

--- a/.github/workflows/build_wheel_manylinux2014.yml
+++ b/.github/workflows/build_wheel_manylinux2014.yml
@@ -72,7 +72,7 @@ jobs:
   upload-pypi:
     needs: linux-wheels-x86-64
     runs-on: ubuntu-latest
-    if: 1 # ${{ github.event_name == 'release' || github.ref == 'refs/heads/main'}}
+    if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/main'}}
     steps:
       - uses: actions/download-artifact@v2
         with:

--- a/.github/workflows/build_wheel_manylinux2014.yml
+++ b/.github/workflows/build_wheel_manylinux2014.yml
@@ -72,7 +72,7 @@ jobs:
   upload-pypi:
     needs: linux-wheels-x86-64
     runs-on: ubuntu-latest
-    if: 0 # ${{ github.event_name == 'release' || github.ref == 'refs/heads/main'}}
+    if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/main'}}
     steps:
       - uses: actions/download-artifact@v2
         with:

--- a/.github/workflows/build_wheel_manylinux2014.yml
+++ b/.github/workflows/build_wheel_manylinux2014.yml
@@ -72,7 +72,7 @@ jobs:
   upload-pypi:
     needs: linux-wheels-x86-64
     runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/main'}}
+    if: 1 # ${{ github.event_name == 'release' || github.ref == 'refs/heads/main'}}
     steps:
       - uses: actions/download-artifact@v2
         with:

--- a/bin/format
+++ b/bin/format
@@ -2,6 +2,7 @@
 
 import argparse
 import json
+import re
 import pathlib
 import subprocess
 import sys
@@ -18,16 +19,28 @@ IGNORE_PATTERN = ["external"]
 
 DEFAULT_CLANG_FORMAT_VERSION=12
 
-BASE_CMD = (f"clang-format", f"-style={json.dumps(CLANG_FMT_STYLE_CFG)}")
+BASE_CMD = (f"clang-format-{DEFAULT_CLANG_FORMAT_VERSION}", f"-style={json.dumps(CLANG_FMT_STYLE_CFG)}")
+
+def parse_version(version_string):
+    version_rgx = "version (\d+)"
+
+    m = re.search(version_rgx, version_string)
+    return int(m.group(1))
 
 def check_bin():
+    command = f"clang-format-{DEFAULT_CLANG_FORMAT_VERSION}"
     try:
-        subprocess.run(
-            (f"clang-format", "--help"), universal_newlines=True, capture_output=True
-        )
+        p = subprocess.run([command, "--version"], 
+        stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
+        version = parse_version(p.stdout)
+
+        if version < DEFAULT_CLANG_FORMAT_VERSION:
+            print(f"Using clang-format version {version}. \
+                    As this is lower than the version used for the CI, \
+                    the CI may fail even after formatting.")
     except FileNotFoundError as exc:
         raise FileNotFoundError(
-            f"clang-format-{DEFAULT_CLANG_FORMAT_VERSION} is not installed or is not in PATH."
+            f"{command} is not installed or is not in PATH."
         ) from exc
 
 

--- a/bin/format
+++ b/bin/format
@@ -18,12 +18,12 @@ IGNORE_PATTERN = ["external"]
 
 DEFAULT_CLANG_FORMAT_VERSION=12
 
-BASE_CMD = (f"clang-format-{DEFAULT_CLANG_FORMAT_VERSION}", f"-style={json.dumps(CLANG_FMT_STYLE_CFG)}")
+BASE_CMD = (f"clang-format", f"-style={json.dumps(CLANG_FMT_STYLE_CFG)}")
 
 def check_bin():
     try:
         subprocess.run(
-            (f"clang-format-{DEFAULT_CLANG_FORMAT_VERSION}", "--help"), universal_newlines=True, capture_output=True
+            (f"clang-format", "--help"), universal_newlines=True, capture_output=True
         )
     except FileNotFoundError as exc:
         raise FileNotFoundError(

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -10,18 +10,18 @@ RUN yum-config-manager --add-repo https://developer.download.nvidia.com/compute/
 RUN mkdir -p -m 0700 ~/.ssh && ssh-keyscan github.com >> ~/.ssh/known_hosts
 
 # Fetch cuQuantum
-RUN wget https://developer.download.nvidia.com/compute/cuquantum/redist/cuquantum/linux-x86_64/cuquantum-linux-x86_64-0.1.0.30-archive.tar.xz && tar xvf ./cuquantum-linux-x86_64-0.1.0.30-archive.tar.xz
+RUN wget https://developer.download.nvidia.com/compute/cuquantum/redist/cuquantum/linux-x86_64/cuquantum-linux-x86_64-22.03.0.40-archive.tar.xz && tar xvf ./cuquantum-linux-x86_64-22.03.0.40-archive.tar.xz
 
 COPY ./  /pennylane-lightning-gpu
 
 # Create venv for each required Python version
 RUN cd /pennylane-lightning-gpu \
     && export PATH=$PATH:/usr/local/cuda/bin \
-    && export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/cuquantum-linux-x86_64-0.1.0.30-archive/lib \
-    && for i in {7..10}; do python3.${i} -m venv pyenv3.${i}; source pyenv3.${i}/bin/activate && python3 -m pip install auditwheel ninja wheel && python3 setup.py build_ext --cuquantum=/cuquantum-linux-x86_64-0.1.0.30-archive --define=CMAKE_CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/g++ --define=ENABLE_CLANG_TIDY=0  && python3 setup.py bdist_wheel && deactivate && rm -rf ./build ; done
+    && export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/cuquantum-linux-x86_64-22.03.0.40-archive/lib \
+    && for i in {7..10}; do python3.${i} -m venv pyenv3.${i}; source pyenv3.${i}/bin/activate && python3 -m pip install auditwheel ninja wheel && python3 setup.py build_ext --cuquantum=/cuquantum-linux-x86_64-22.03.0.40-archive --define=CMAKE_CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/g++ --define=ENABLE_CLANG_TIDY=0  && python3 setup.py bdist_wheel && deactivate && rm -rf ./build ; done
 
 RUN cd /pennylane-lightning-gpu \
     && export PATH=$PATH:/usr/local/cuda/bin \
-    && export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/cuquantum-linux-x86_64-0.1.0.30-archive/lib \
+    && export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/cuquantum-linux-x86_64-22.03.0.40-archive/lib \
     && source pyenv3.10/bin/activate \
     && for i in $(ls ./dist); do /pennylane-lightning-gpu/docker/auditwheel repair -w /wheelhouse /pennylane-lightning-gpu/dist/$i; done

--- a/pennylane_lightning_gpu/_version.py
+++ b/pennylane_lightning_gpu/_version.py
@@ -16,4 +16,4 @@
    Version number (major.minor.patch[-label])
 """
 
-__version__ = "0.2.0-dev0"
+__version__ = "0.22.0"

--- a/pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp
+++ b/pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp
@@ -123,7 +123,8 @@ class StateVectorCudaManaged
 
     {
         BaseType::initSV();
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecCreate(&handle));
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecCreate(
+            /* custatevecHandle_t* */ &handle));
     };
 
     StateVectorCudaManaged(const CFP_t *gpu_data, size_t length)
@@ -143,7 +144,8 @@ class StateVectorCudaManaged
     // StateVectorCudaManaged(StateVectorCudaManaged &&other) = delete;
 
     ~StateVectorCudaManaged() {
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecDestroy(handle));
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecDestroy(
+            /* custatevecHandle_t */ handle));
     }
 
     /**
@@ -537,9 +539,16 @@ class StateVectorCudaManaged
             });
 
         PL_CUSTATEVEC_IS_SUCCESS(custatevecAbs2SumArray(
-            handle, BaseType::getData(), data_type, BaseType::getNumQubits(),
-            probabilities.data(), wires_int.data(), wires_int.size(),
-            maskBitString, maskOrdering, maskLen));
+            /* custatevecHandle_t */ handle,
+            /* const void* */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ BaseType::getNumQubits(),
+            /* double* */ probabilities.data(),
+            /* const int32_t* */ wires_int.data(),
+            /* const uint32_t */ wires_int.size(),
+            /* const int32_t* */ maskBitString,
+            /* const int32_t* */ maskOrdering,
+            /* const uint32_t */ maskLen));
 
         return probabilities;
     }
@@ -620,10 +629,18 @@ class StateVectorCudaManaged
         }
         const auto local_angle = (use_adjoint) ? param / 2 : -param / 2;
 
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyExp(
-            handle, BaseType::getData(), data_type, nIndexBits, local_angle,
-            pauli_enums.data(), tgtsInt.data(), tgts.size(), ctrlsInt.data(),
-            nullptr, ctrls.size()));
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyPauliRotation(
+            /* custatevecHandle_t */ handle,
+            /* void* */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* double */ local_angle,
+            /* const custatevecPauli_t* */ pauli_enums.data(),
+            /* const int32_t* */ tgtsInt.data(),
+            /* const uint32_t */ tgts.size(),
+            /* const int32_t* */ ctrlsInt.data(),
+            /* const int32_t* */ nullptr,
+            /* const uint32_t */ ctrls.size()));
     }
 
     /**
@@ -675,7 +692,7 @@ class StateVectorCudaManaged
             /* custatevecHandle_t */ handle,
             /* cudaDataType_t */ data_type,
             /* const uint32_t */ nIndexBits,
-            /* const void* */ matrix.data(), // double-check
+            /* const void* */ matrix,
             /* cudaDataType_t */ data_type,
             /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
             /* const int32_t */ use_adjoint,
@@ -696,14 +713,14 @@ class StateVectorCudaManaged
             /* void* */ BaseType::getData(),
             /* cudaDataType_t */ data_type,
             /* const uint32_t */ nIndexBits,
-            /* const void* */ matrix.data(), // double-check
+            /* const void* */ matrix,
             /* cudaDataType_t */ data_type,
             /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
             /* const int32_t */ use_adjoint,
             /* const int32_t* */ tgtsInt.data(),
             /* const uint32_t */ tgts.size(),
             /* const int32_t* */ ctrlsInt.data(),
-            /* const int32_t*  */ nullptr, // double-check
+            /* const int32_t* */ nullptr,
             /* const uint32_t */ ctrls.size(),
             /* custatevecComputeType_t */ compute_type,
             /* void* */ extraWorkspace,
@@ -787,7 +804,7 @@ class StateVectorCudaManaged
             /* const int32_t* */ tgtsInt.data(),
             /* const uint32_t */ tgts.size(),
             /* const int32_t* */ ctrlsInt.data(),
-            /* const int32_t*  */ nullptr, // double-check
+            /* const int32_t* */ nullptr,
             /* const uint32_t */ ctrls.size(),
             /* custatevecComputeType_t */ compute_type,
             /* void* */ extraWorkspace,
@@ -916,7 +933,7 @@ class StateVectorCudaManaged
             /* custatevecHandle_t */ handle,
             /* cudaDataType_t */ data_type,
             /* const uint32_t */ nIndexBits,
-            /* const void* */ matrix.data(), // double-check
+            /* const void* */ matrix,
             /* cudaDataType_t */ data_type,
             /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
             /* const uint32_t */ tgtsInt.size(),
@@ -939,7 +956,7 @@ class StateVectorCudaManaged
             /* void* */ &expect,
             /* cudaDataType_t */ data_type,
             /* double* */ nullptr,
-            /* const void* */ matrix.data(), // double-check
+            /* const void* */ matrix,
             /* cudaDataType_t */ data_type,
             /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
             /* const int32_t* */ tgtsInt.data(),

--- a/pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp
+++ b/pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp
@@ -6,7 +6,7 @@
 
 #include <cuComplex.h> // cuDoubleComplex
 #include <cuda.h>
-#include <custatevec.h>
+#include <custatevec.h> // custatevecApplyMatrix
 
 #include "Error.hpp"
 #include "StateVectorCudaBase.hpp"
@@ -620,7 +620,7 @@ class StateVectorCudaManaged
         }
         const auto local_angle = (use_adjoint) ? param / 2 : -param / 2;
 
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyPauliRotation(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyExp(
             handle, BaseType::getData(), data_type, nIndexBits, local_angle,
             pauli_enums.data(), tgtsInt.data(), tgts.size(), ctrlsInt.data(),
             nullptr, ctrls.size()));
@@ -672,9 +672,17 @@ class StateVectorCudaManaged
 
         // check the size of external workspace
         PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrixGetWorkspaceSize(
-            handle, data_type, nIndexBits, matrix, data_type,
-            CUSTATEVEC_MATRIX_LAYOUT_ROW, use_adjoint, tgts.size(),
-            ctrls.size(), compute_type, &extraWorkspaceSizeInBytes));
+            /* custatevecHandle_t */ handle,
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* const void* */ matrix.data(), // double-check
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const int32_t */ use_adjoint,
+            /* const uint32_t */ tgts.size(),
+            /* const uint32_t */ ctrls.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* size_t* */ &extraWorkspaceSizeInBytes));
 
         // allocate external workspace if necessary
         if (extraWorkspaceSizeInBytes > 0) {
@@ -683,11 +691,23 @@ class StateVectorCudaManaged
         }
 
         // apply gate
-        (custatevecApplyMatrix(
-            handle, BaseType::getData(), data_type, nIndexBits, matrix,
-            data_type, CUSTATEVEC_MATRIX_LAYOUT_ROW, use_adjoint,
-            tgtsInt.data(), tgts.size(), ctrlsInt.data(), ctrls.size(), nullptr,
-            compute_type, extraWorkspace, extraWorkspaceSizeInBytes));
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrix(
+            /* custatevecHandle_t */ handle,
+            /* void* */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* const void* */ matrix.data(), // double-check
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const int32_t */ use_adjoint,
+            /* const int32_t* */ tgtsInt.data(),
+            /* const uint32_t */ tgts.size(),
+            /* const int32_t* */ ctrlsInt.data(),
+            /* const int32_t*  */ nullptr, // double-check
+            /* const uint32_t */ ctrls.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* void* */ extraWorkspace,
+            /* size_t */ extraWorkspaceSizeInBytes));
         if (extraWorkspaceSizeInBytes)
             PL_CUDA_IS_SUCCESS(cudaFree(extraWorkspace));
     }
@@ -736,9 +756,17 @@ class StateVectorCudaManaged
 
         // check the size of external workspace
         PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrixGetWorkspaceSize(
-            handle, data_type, nIndexBits, matrix.data(), data_type,
-            CUSTATEVEC_MATRIX_LAYOUT_ROW, use_adjoint, tgts.size(),
-            ctrls.size(), compute_type, &extraWorkspaceSizeInBytes));
+            /* custatevecHandle_t */ handle,
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* const void* */ matrix.data(),
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const int32_t */ use_adjoint,
+            /* const uint32_t */ tgts.size(),
+            /* const uint32_t */ ctrls.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* size_t* */ &extraWorkspaceSizeInBytes));
 
         // allocate external workspace if necessary
         if (extraWorkspaceSizeInBytes > 0) {
@@ -748,10 +776,22 @@ class StateVectorCudaManaged
 
         // apply gate
         PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrix(
-            handle, BaseType::getData(), data_type, nIndexBits, matrix.data(),
-            data_type, CUSTATEVEC_MATRIX_LAYOUT_ROW, use_adjoint,
-            tgtsInt.data(), tgts.size(), ctrlsInt.data(), ctrls.size(), nullptr,
-            compute_type, extraWorkspace, extraWorkspaceSizeInBytes));
+            /* custatevecHandle_t */ handle,
+            /* void* */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* const void* */ matrix.data(),
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const int32_t */ use_adjoint,
+            /* const int32_t* */ tgtsInt.data(),
+            /* const uint32_t */ tgts.size(),
+            /* const int32_t* */ ctrlsInt.data(),
+            /* const int32_t*  */ nullptr, // double-check
+            /* const uint32_t */ ctrls.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* void* */ extraWorkspace,
+            /* size_t */ extraWorkspaceSizeInBytes));
         if (extraWorkspaceSizeInBytes)
             PL_CUDA_IS_SUCCESS(cudaFree(extraWorkspace));
     }
@@ -799,10 +839,17 @@ class StateVectorCudaManaged
             compute_type = CUSTATEVEC_COMPUTE_32F;
         }
 
+        // check the size of external workspace
         PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectationGetWorkspaceSize(
-            handle, data_type, nIndexBits, matrix.data(), data_type,
-            CUSTATEVEC_MATRIX_LAYOUT_ROW, tgts.size(), compute_type,
-            &extraWorkspaceSizeInBytes));
+            /* custatevecHandle_t */ handle,
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* const void* */ matrix.data(),
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const uint32_t */ tgts.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* size_t* */ &extraWorkspaceSizeInBytes));
 
         if (extraWorkspaceSizeInBytes > 0) {
             PL_CUDA_IS_SUCCESS(
@@ -813,11 +860,21 @@ class StateVectorCudaManaged
 
         // compute expectation
         PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectation(
-            handle, BaseType::getData(), data_type, nIndexBits, &expect,
-            data_type, nullptr, matrix.data(), data_type,
-            CUSTATEVEC_MATRIX_LAYOUT_ROW, tgtsInt.data(), tgts.size(),
-            compute_type, extraWorkspace, extraWorkspaceSizeInBytes));
-
+            /* custatevecHandle_t */ handle,
+            /* void* */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* void* */ &expect,
+            /* cudaDataType_t */ data_type,
+            /* double* */ nullptr,
+            /* const void* */ matrix.data(),
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const int32_t* */ tgtsInt.data(),
+            /* const uint32_t */ tgts.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* void* */ extraWorkspace,
+            /* size_t */ extraWorkspaceSizeInBytes));
         if (extraWorkspaceSizeInBytes)
             PL_CUDA_IS_SUCCESS(cudaFree(extraWorkspace));
         return expect;
@@ -854,10 +911,17 @@ class StateVectorCudaManaged
             compute_type = CUSTATEVEC_COMPUTE_32F;
         }
 
+        // check the size of external workspace
         PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectationGetWorkspaceSize(
-            handle, data_type, nIndexBits, matrix, data_type,
-            CUSTATEVEC_MATRIX_LAYOUT_ROW, tgtsInt.size(), compute_type,
-            &extraWorkspaceSizeInBytes));
+            /* custatevecHandle_t */ handle,
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* const void* */ matrix.data(), // double-check
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const uint32_t */ tgtsInt.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* size_t* */ &extraWorkspaceSizeInBytes));
 
         if (extraWorkspaceSizeInBytes > 0) {
             PL_CUDA_IS_SUCCESS(
@@ -868,10 +932,21 @@ class StateVectorCudaManaged
 
         // compute expectation
         PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectation(
-            handle, BaseType::getData(), data_type, nIndexBits, &expect,
-            data_type, nullptr, matrix, data_type, CUSTATEVEC_MATRIX_LAYOUT_ROW,
-            tgtsInt.data(), tgtsInt.size(), compute_type, extraWorkspace,
-            extraWorkspaceSizeInBytes));
+            /* custatevecHandle_t */ handle,
+            /* void* */ BaseType::getData(),
+            /* cudaDataType_t */ data_type,
+            /* const uint32_t */ nIndexBits,
+            /* void* */ &expect,
+            /* cudaDataType_t */ data_type,
+            /* double* */ nullptr,
+            /* const void* */ matrix.data(), // double-check
+            /* cudaDataType_t */ data_type,
+            /* custatevecMatrixLayout_t */ CUSTATEVEC_MATRIX_LAYOUT_ROW,
+            /* const int32_t* */ tgtsInt.data(),
+            /* const uint32_t */ tgtsInt.size(),
+            /* custatevecComputeType_t */ compute_type,
+            /* void* */ extraWorkspace,
+            /* size_t */ extraWorkspaceSizeInBytes));
 
         if (extraWorkspaceSizeInBytes)
             PL_CUDA_IS_SUCCESS(cudaFree(extraWorkspace));

--- a/pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp
+++ b/pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp
@@ -6,7 +6,7 @@
 
 #include <cuComplex.h> // cuDoubleComplex
 #include <cuda.h>
-#include <custatevec.h> // custatevecApplyMatrix
+#include <custatevec.h>
 
 #include "Error.hpp"
 #include "StateVectorCudaBase.hpp"
@@ -620,7 +620,7 @@ class StateVectorCudaManaged
         }
         const auto local_angle = (use_adjoint) ? param / 2 : -param / 2;
 
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyExp(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyPauliRotation(
             handle, BaseType::getData(), data_type, nIndexBits, local_angle,
             pauli_enums.data(), tgtsInt.data(), tgts.size(), ctrlsInt.data(),
             nullptr, ctrls.size()));
@@ -671,7 +671,7 @@ class StateVectorCudaManaged
         }
 
         // check the size of external workspace
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrix_bufferSize(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrixGetWorkspaceSize(
             handle, data_type, nIndexBits, matrix, data_type,
             CUSTATEVEC_MATRIX_LAYOUT_ROW, use_adjoint, tgts.size(),
             ctrls.size(), compute_type, &extraWorkspaceSizeInBytes));
@@ -735,7 +735,7 @@ class StateVectorCudaManaged
         }
 
         // check the size of external workspace
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrix_bufferSize(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecApplyMatrixGetWorkspaceSize(
             handle, data_type, nIndexBits, matrix.data(), data_type,
             CUSTATEVEC_MATRIX_LAYOUT_ROW, use_adjoint, tgts.size(),
             ctrls.size(), compute_type, &extraWorkspaceSizeInBytes));
@@ -799,7 +799,7 @@ class StateVectorCudaManaged
             compute_type = CUSTATEVEC_COMPUTE_32F;
         }
 
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecExpectation_bufferSize(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectationGetWorkspaceSize(
             handle, data_type, nIndexBits, matrix.data(), data_type,
             CUSTATEVEC_MATRIX_LAYOUT_ROW, tgts.size(), compute_type,
             &extraWorkspaceSizeInBytes));
@@ -812,7 +812,7 @@ class StateVectorCudaManaged
         CFP_t expect;
 
         // compute expectation
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecExpectation(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectation(
             handle, BaseType::getData(), data_type, nIndexBits, &expect,
             data_type, nullptr, matrix.data(), data_type,
             CUSTATEVEC_MATRIX_LAYOUT_ROW, tgtsInt.data(), tgts.size(),
@@ -854,7 +854,7 @@ class StateVectorCudaManaged
             compute_type = CUSTATEVEC_COMPUTE_32F;
         }
 
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecExpectation_bufferSize(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectationGetWorkspaceSize(
             handle, data_type, nIndexBits, matrix, data_type,
             CUSTATEVEC_MATRIX_LAYOUT_ROW, tgtsInt.size(), compute_type,
             &extraWorkspaceSizeInBytes));
@@ -867,7 +867,7 @@ class StateVectorCudaManaged
         CFP_t expect;
 
         // compute expectation
-        PL_CUSTATEVEC_IS_SUCCESS(custatevecExpectation(
+        PL_CUSTATEVEC_IS_SUCCESS(custatevecComputeExpectation(
             handle, BaseType::getData(), data_type, nIndexBits, &expect,
             data_type, nullptr, matrix, data_type, CUSTATEVEC_MATRIX_LAYOUT_ROW,
             tgtsInt.data(), tgtsInt.size(), compute_type, extraWorkspace,

--- a/pennylane_lightning_gpu/src/tests/CMakeLists.txt
+++ b/pennylane_lightning_gpu/src/tests/CMakeLists.txt
@@ -28,24 +28,24 @@ list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/contrib)
 include(CTest)
 include(Catch)
 
-add_executable(runner runner_main.cpp)
+add_executable(runner_gpu runner_main.cpp)
 if(ENABLE_OPENMP)
     find_package(OpenMP REQUIRED)
 	  target_link_libraries(runner INTERFACE OpenMP::OpenMP_CXX)
 endif()
-target_link_libraries(runner PUBLIC pennylane_lightning_gpu Catch2::Catch2 lightning_simulator)
+target_link_libraries(runner_gpu PUBLIC pennylane_lightning_gpu Catch2::Catch2 lightning_simulator)
 
-target_sources(runner PRIVATE Test_StateVectorCudaManaged_NonParam.cpp 
+target_sources(runner_gpu PRIVATE Test_StateVectorCudaManaged_NonParam.cpp 
                               Test_StateVectorCudaManaged_Param.cpp 
                               Test_AdjointDiffGPU.cpp 
                               Test_GateCache.cpp 
 )
 
-target_compile_options(runner PRIVATE "$<$<CONFIG:DEBUG>:-Wall>")
+target_compile_options(runner_gpu PRIVATE "$<$<CONFIG:DEBUG>:-Wall>")
 
 if(ENABLE_NATIVE)
     message(STATUS "ENABLE_NATIVE is ON. Use -march=native for cpptests.")
-    target_compile_options(runner PRIVATE -march=native)
+    target_compile_options(runner_gpu PRIVATE -march=native)
 endif()
 
-catch_discover_tests(runner)
+catch_discover_tests(runner_gpu)

